### PR TITLE
test: add offset tests

### PR DESCRIPTION
**Pull request by llugin**
Will not pass without merging #182 first

---
This change is [Reviewable](https://reviewable.io/reviews/pmem/vmemcache/188)

### CODE REVIEW

**Comment by llugin** [comment]
rebased

File: tests/vmemcache_test_basic.c [hunk 1]
```diff
@@ -1150,6 +1150,108 @@ test_get_with_offset(const char *dir)
 	vmemcache_delete(cache);
 }
 
+/* test_offsets test case parameters */
+struct offset_tc {
+	char *vbuf;
+	ssize_t vbuf_size;
+	ssize_t offset;
```
Comment (marcinslusarz) [inline tests/vmemcache_test_basic.c]: Why are those 2 signed?

File: tests/vmemcache_test_basic.c [hunk 2]
```diff
@@ -1150,6 +1150,109 @@ test_get_with_offset(const char *dir)
...
+};
+
+/*
+ * run_offset_tc -- (internal) run single get with offset test case
+ */
+static void
+run_offset_tc(struct offset_tc *tc, VMEMcache *cache, const char *key,
+			size_t ksize, const char *val, size_t val_size)
+{
+		size_t vsize;
```
Comment (marcinslusarz) [inline tests/vmemcache_test_basic.c]: indentation

File: tests/vmemcache_test_basic.c [hunk 3]
```diff
@@ -1150,6 +1150,109 @@ test_get_with_offset(const char *dir)
...
+	VMEMcache *cache = vmemcache_new(dir, VMEMCACHE_MIN_POOL,
+		VMEMCACHE_MIN_EXTENT, policy);
+	if (cache == NULL)
+		UT_FATAL("vmemcache_new: %s", vmemcache_errormsg());
+
+	const char *key = "KEY";
+	size_t ksize = strlen(key) + 1;
+
+	char val[val_size];
+	for (unsigned i = 0; i < val_size - 1; i++) {
```
Comment (marcinslusarz) [inline tests/vmemcache_test_basic.c]: {} -> /dev/null

File: tests/vmemcache_test_basic.c (hunk 2 above)
Comment (llugin) [reply tests/vmemcache_test_basic.c]: Done

File: tests/vmemcache_test_basic.c (hunk 3 above)
Comment (llugin) [reply tests/vmemcache_test_basic.c]: Done

**Pull request merged**